nifti2/nifti_tool.c

```diff
@@ -402,15 +401,15 @@ int process_opts( int argc, char * argv[], nt_opts * opts )
       /* begin normal execution options... */
       else if( ! strcmp(argv[ac], "-add_ext") )
       {
+         int new_ecode;
          ac++;
          CHECK_NEXT_OPT(ac, argc, "-add_ext");
-         errno = 0;
-         long long new_ecode = strtoll(argv[ac], NULL, 10);
-         if ((new_ecode == 0 && errno != 0) || new_ecode <= INT_MIN || new_ecode >= INT_MAX) {
-           fprintf(stderr,"** invalid -add_ext parameter\n");
+         new_ecode = atoi(argv[ac]);
```

Comment:
Why revert to atoi?  It is obsolete. It results in undefined behaviour if the string can't fit into an int.

See for example:

https://wiki.sei.cmu.edu/confluence/display/c/MSC24-C.+Do+not+use+deprecated+or+obsolescent+functions

Reply:
Hmmm, it looks like it is being pushed towards obsolescence, but of course, will never be.  But I believe it returns 0 on error, not an undefined value.  I wanted to remove the unnecessary use of long long (which is a very annoying type, since it's actual type varies).  The ecode is question only has a small range of acceptable values, which will be checked.  Using atoi() is not important, but going to large types is ugly.

Reply:
>But I believe it returns 0 on error, not an undefined value

[C99 7.20.1](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) "If the value of the result cannot be represented, the behavior is undefined."

> ... use of long long (which is a very annoying type, since it's actual type varies)

That's true of all the types though, even the `int` that `atoi` uses.

>but going to large types is ugly

Ugly is preferable to undefined behaviour though.